php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -151,6 +162,19 @@ public function __construct($desc = NULL)
         }
     }
 
+    /**
+     * @ignore
+     */
+    private function initWithDescriptor(Descriptor $desc)
+    {
+        $this->desc = $desc;
```

Comment:
It's not needed because it's an internal class and the only times it's instantiated is with a descriptor, so this'll be executed. 

Essentially I took the behavior but made it generic. Checking the class in the constructor is not as clean, and this results in the same behavior.